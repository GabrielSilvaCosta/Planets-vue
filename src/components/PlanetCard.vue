<template>
    <div class="planet-card">
      <h3>{{ planetName }}</h3>
      <div class="image-container">
        <img :src="planetImage" :alt="planetName" class="planet-image" />
      </div>
    </div>
  </template>
  
  <script>
  export default {
    name: "PlanetCard",
    props: {
      planetName: {
        type: String,
        required: true,
      },
      planetImage: {
        type: String,
        required: true,
      },
    },
  };
  </script>
  
  <style scoped>
  .planet-card {
    text-align: center;
    margin-bottom: 20px;
    padding: 20px;
    border: 1px solid #ccc;
    border-radius: 10px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    
  }
  
  .image-container {
    display: flex;
    justify-content: center;
    margin-bottom: 10px;
    margin-top: 20px;
    

  }
  
  .planet-image {
    width: 200px;
    height: 200px;
    object-fit: cover;
    border-radius: 10px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    
  }
  
  h3 {
    margin: 0;
    font-size: 24px;
    font-weight: bold;
    text-transform: uppercase;
    letter-spacing: 2px;
    margin-bottom: 10px;
  }
  
  </style>
  