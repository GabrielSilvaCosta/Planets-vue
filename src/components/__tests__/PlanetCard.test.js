import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import PlanetCard from '../PlanetCard.vue';

describe('PlanetCard', () => {
    it('should render correctly', () => {
        const wrapper = mount(PlanetCard, {
            props: {
                planet: {
                    name: 'Mercurio',
                    image: 'https://via.placeholder.com/150',
                },
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
})