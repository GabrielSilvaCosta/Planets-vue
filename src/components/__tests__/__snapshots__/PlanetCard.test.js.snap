// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlanetCard > should render correctly 1`] = `
"<div data-v-1ed930a5="" class="planet-card" planet="[object Object]">
  <h3 data-v-1ed930a5=""></h3>
  <div data-v-1ed930a5="" class="image-container"><img data-v-1ed930a5="" class="planet-image"></div>
</div>"
`;
